lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -505,6 +505,7 @@ def links_for(serializer)
       # prs:
       #   https://github.com/rails-api/active_model_serializers/pull/1041
       def pagination_links_for(serializer)
+        return {} unless instance_options.key?(:serialization_context)
```

Comment:
ref:

```ruby

        if is_collection && serializer.paginated?
          hash[:links] ||= {}
          hash[:links].update(pagination_links_for(serializer))
        end
```

I did at first, but then I was thinking it's really tightly coupled to the paginationlinks object and it's intent isn't as obvious in the conditional.

But really we shouldn't let an empty serialization_context get in that far.  This is a quick fix, but the better fix is at the boundary, I think